packages/gestalt/src/Masonry.js

```diff
@@ -496,15 +500,19 @@ export default class Masonry<T: { ... }> extends ReactComponent<Props<T>, State<
           style={{ height: 0, width: '100%' }}
         >
           {items
+            // this is here to filter out falsy values
             .filter((item) => item)
             .map((item, i) => (
               <div // keep this in sync with renderMasonryComponent
                 className="static"
                 data-grid-item
                 key={i}
                 ref={(el) => {
+                  // purposely not checking for layout === 'serverRenderedFlexible' here
                   if (el && !(flexible || layout === 'flexible')) {
-                    // only measure flexible items on client
+                    // if we're hydrating from the server, we should only measure items on the initial render pass
+                    // if we're not rendering a flexible layout.  "serverRenderedFlexible" is an exception because we assume
+                    // that the caller has added the proper CSS to ensure the layout is correct during server render
```

Comment:
Should we add a similar note in the docs?

Reply:
the `layout` section of the docs actually seems broken
![image](https://user-images.githubusercontent.com/6487551/165625602-67128168-e1ed-4489-8513-503d268d737d.png)

will take a look at that separately and move this over when that's fixed.  otherwise its a bit confusing imo